.github/workflows/update-authorized-users.yaml

```
@@ -1,22 +1,23 @@
 name: Update authorized users
 
 on:
-  workflow_dispatch:
-  schedule:
-    - cron: '30 21 * * 3'
+  pull_request_target:
+    types:
+      - closed
+    branches: 
+      - main
+    paths:
+      - 'model-metadata/*.yaml'
```

Perhaps simpler to do on a push to main that changes `'model-metadata/*.yaml'`?

I believe the syntax is as follows (though you should double-check)

```suggestion
  push:
    branches: [main]
    paths: ['model-metadata/*.yaml']
```

Note that a merged PR counts as a "push"